jax/experimental/jet.py

```diff
@@ -12,6 +12,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+r"""Jet is an experimental module for higher-order automatic differentiation
+  that does not rely on repeated first-order automatic differentiation.
+
+  How? Through the propagation of truncated Taylor polynomials.
+  Let :math:`x: \mathbb{R} \rightarrow \mathbb{R}^n` be the Taylor polynomial
+
+  .. math:: x(t) := x_0 + x_1 t + \frac{x_2}{2!} t^2 + ... + \frac{1}{d!} x_d t^d.
+
+  For some function :math:`f: \mathbb{R}^n \rightarrow \mathbb{R}^m`,
+  you can use :func:`jet` to compute the
+  Taylor polynomial
+
+  .. math:: y(t) := y_0 + y_1 t + \frac{y_2}{2!} t^2 + ... + \frac{1}{d!} y_d t^d.
+
+  corresponding to :math:`f(x(\cdot)):\mathbb{R} \rightarrow \mathbb{R}^m`,
+  and relates to :math:`x(t)` and :math:`f(z)` through
+
+  .. math::
+    y_k = \frac{d^k}{dt^k}f(x(t_0)), \quad k=0,...,d.
+
+  More specifically, :func:`jet` computes
+
+  .. math::
+    y_0, (y_1, . . . , y_d) = \texttt{jet} (f, x_0, (x_1, . . . , x_d))
+
+  and can thus be used for high-order
+  automatic differentiation of :math:`f`.
+  Details are explained in
+  `this paper <https://openreview.net/forum?id=SkxEF3FNPH>`__.
```

Comment:
The content of this docstring closely follows the exposition in https://openreview.net/pdf?id=SkxEF3FNPH. It seems to be more maths than other, comparable module docstrings, but I figured it is useful to establish the variables for easier reference of what `jet` does.

Reply:
Thanks for pushing on this! 

I don't recommend following the math or notation in that workshop paper. I recommend working from [these notes](https://github.com/google/jax/files/6717197/jet.pdf). In particular, the differentiation notation is different, with no \frac. We should link to that pdf as well (instead of openreview).

Reply:
Good to know, thanks! I updated the link as well as the notation to be more in line with the notes you linked.